From 83aa893b04e9e70f7f51db7d2d71c8d7c81e4f44 Mon Sep 17 00:00:00 2001
From: Ren Jianing <jianing.ren@rock-chips.com>
Date: Fri, 25 Sep 2020 09:50:11 +0800
Subject: [PATCH] usb: gadget: legacy: audio: fix error parameter of sizeof

This patch fixes the issue that a pointer is used as the parameter
of sizeof.

Fixes: 9de5fa6df194 (FROMLIST: usb: gadget: f_uac*: Reduce code duplication)
Signed-off-by: Ren Jianing <jianing.ren@rock-chips.com>
Change-Id: I66bfc58e71c498dd8a82f9b5045f851585e1f441
---
 drivers/usb/gadget/legacy/audio.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/legacy/audio.c b/drivers/usb/gadget/legacy/audio.c
index f792afd25d22..104d1c4ce7f9 100644
--- a/drivers/usb/gadget/legacy/audio.c
+++ b/drivers/usb/gadget/legacy/audio.c
@@ -225,7 +225,7 @@ static int audio_bind(struct usb_composite_dev *cdev)
 #endif
 
 #if !defined(CONFIG_GADGET_UAC1) || !defined(CONFIG_GADGET_UAC1_LEGACY)
-	memset(uac_opts, 0x0, sizeof(uac_opts));
+	memset(uac_opts, 0x0, sizeof(*uac_opts));
 	uac_opts->p_chmask = p_chmask;
 	uac_opts->p_srate[0] = p_srate;
 	uac_opts->p_srate_active = p_srate;
-- 
2.35.3

